### PR TITLE
ghc 9.2

### PR DESCRIPTION
**Pull request by robrix**
This PR adds support for ghc 9.2.

### CODE REVIEW

File: src/Data/ProtoLens/JSONPB/Class.hs [hunk 1]
```diff
@@ -141,10 +135,10 @@ class Monoid m =>
   pair :: ToJSONPB v => Text -> v -> Options -> m
 
 instance KeyValuePB A.Series where
-  pair k v opts = E.pair k (toEncodingPB v opts)
+  pair k v opts = E.pair (fromString (unpack k)) (toEncodingPB v opts)
```
Comment (robrix) [inline src/Data/ProtoLens/JSONPB/Class.hs:138]: This is gross, but `aeson` now has an opaque `Key` type which we have to construct like this in order to support older versions of the library (and thus avoid tying ourselves to `aeson`'s compiler support, not to mention having to figure out the correct lower bound 😄).

File: src/Data/ProtoLens/JSONPB/Class.hs [hunk 2]
```diff
@@ -1,4 +1,8 @@
-{-# LANGUAGE FlexibleInstances, OverloadedLists, RecordWildCards, ScopedTypeVariables, ViewPatterns #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE ViewPatterns #-}
```
Comment (robrix) [inline src/Data/ProtoLens/JSONPB/Class.hs:5]: All formatting changes are c/o `stylish-haskell`.

File: proto-lens-jsonpb.cabal [hunk 3]
```diff
@@ -15,9 +15,9 @@ category:       Web
 build-type:     Simple
 extra-source-files: README.md
 
-tested-with:    GHC == 8.6.5
-              , GHC == 8.8.1
-              , GHC == 8.10.1
+tested-with:    GHC == 8.10.1
+              , GHC == 9.0.1
+              , GHC == 9.2.1
```
Comment (robrix) [inline proto-lens-jsonpb.cabal:20]: We haven't dropped support for older compilers, but we're not going to test against them in CI. They're officially in YMMV territory now.

**Comment by robrix** [comment]
(I'll also bump versions and do a new release once accepted.)

**Review by patrickt — APPROVED** [review]

**Comment by robrix** [comment]
https://hackage.haskell.org/package/proto-lens-jsonpb-0.2.1/candidate

**Comment by robrix** [comment]
Released! https://hackage.haskell.org/package/proto-lens-jsonpb-0.2.1

**Comment by robrix** [comment]
@tclem: like you, I had been under the impression that me being a collaborator would mean that I could change branch protection rules; as it happens I don't have access to any settings. I'd like to solve this properly, but in the interests of expedience, could you please make the 9.2 build the only required one?

**Comment by tclem** [comment]
np, done!

**Comment by robrix** [comment]
you're the best @tclem ❤️

**Pull request merged**